mambaSharedFramework/InterstitialTagBuilder.swift

```diff
@@ -0,0 +1,293 @@
+        
+        return self
+    }
+    
+    /// Specifies how the interstitial is styled on the timeline
+    ///
+    /// - Parameter style: `HLSInterstitialTimelineStyle` type
+    ///
+    /// - Returns: an instance of the builder
+    @discardableResult
+    public func withTimelineStyle(_ style: HLSInterstitialTimelineStyle) -> Self {
+        self.timelineStyle = style
+        
+        return self
+    }
+    
+    /// Describes how the interstitial occupies the content timeline
+    ///
+    /// - Parameter occupation: `HLSInterstitialTimelineOccupation` type
+    ///
+    /// - Returns: an instance of the builder
+    @discardableResult
+    public func withTimelineOccupation(_ occupation: HLSInterstitialTimelineOccupation) -> Self {
+        self.timelineOccupation = occupation
+        
+        return self
+    }
+    
+    /// Indicates if the interstitial content varies or stays the same during a shared watching activity
+    ///
+    /// - Parameter variation: `Bool` indicating if there's variation
+    ///
+    /// - Returns: an instance of the builder
+    @discardableResult
+    public func withContentVariation(_ variation: Bool) -> Self {
+        self.contentMayVary = variation
+        
+        return self
+    }
+    
+    /// Specifies client attributes describing the interstitial
+    ///
+    /// - Parameter attributes: a map of `[String: LosslessStringConvertible]` describing the attributes
+    ///
+    /// - Returns: an instance of the builder
+    @discardableResult
+    public func withClientAttributes(_ attributes: [String: LosslessStringConvertible]) -> Self {
+        self.clientAttributes = attributes
+        
+        return self
+    }
+    
+    /// Builds the DateRange tag utilizing the configured HLS interstitial properties
+    ///
+    /// - Returns: `HLSTag`
+    public func buildTag() -> HLSTag {
+        
+        var hlsTagDictionary = HLSTagDictionary()
+        
+        hlsTagDictionary[PantosValue.id.rawValue] = HLSValueData(value: id, quoteEscaped: true)
+        let startDateString = String.DateFormatter.iso8601MS.string(from: startDate)
+        hlsTagDictionary[PantosValue.startDate.rawValue] = HLSValueData(value: startDateString,
+                                                                        quoteEscaped: true)
+        hlsTagDictionary[PantosValue.classAttribute.rawValue] = HLSValueData(value: classId,
+                                                                             quoteEscaped: true)
+        
+        if let assetUri {
+            hlsTagDictionary[PantosValue.assetUri.rawValue] = HLSValueData(value: assetUri, quoteEscaped: true)
+        }
+        
+        if let assetList {
+            hlsTagDictionary[PantosValue.assetList.rawValue] = HLSValueData(value: assetList, quoteEscaped: true)
+        }
+        
+        if let duration {
+            hlsTagDictionary[PantosValue.duration.rawValue] = HLSValueData(value: String(duration), quoteEscaped: true)
```

Comment:
Should this be `quoteEscaped`? Doesn't this imply the attribute gets created like `DURATION="60.0"` (for example)?

```suggestion
            hlsTagDictionary[PantosValue.duration.rawValue] = HLSValueData(value: String(duration), quoteEscaped: false)
```

Same question for `resumeOffset` and `playoutLimit` which are defined as "signed-decimal-floating-point" and "decimal-floating-point" respectively.

Reply:
yeah this was a miss. I've updated it to use `false` for those decimal values.